src/java.base/share/classes/sun/security/rsa/RSAPublicKeyImpl.java

```diff
@@ -217,31 +220,37 @@ protected Object writeReplace() throws java.io.ObjectStreamException {
                         getEncoded());
     }
 
-    private long nativeRSAKey = 0x0;
-
-    /**
-     * Get native RSA Public Key context pointer.
-     * Create native context if uninitialized.
-     */
-    protected long getNativePtr() {
-        if (nativeRSAKey != 0x0) {
-            return nativeRSAKey;
-        }
-
+    private long RSAPrivateKey_generate() {
```

Comment:
The name of this method is misleading: it generates the public, key - not the private key.